plutus-benchmark/flat/Codec.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PackageImports    #-}
+
+module Codec ( Codec(..)
+             , flatCodec
+             , cborCodec
+             , withZlib
+             , withPureZlib
+             , codecs
+             ) where
+
+import           Control.DeepSeq                     (NFData, force)
+import qualified Data.ByteString                     as BS
+import qualified Data.ByteString.Lazy                as LBS
+import           Data.Either                         (fromRight)
+import           Data.Text                           (Text)
+
+import           Language.PlutusCore                 (Name (..), Unique (..))
+import           Language.PlutusCore.Universe
+import           Language.UntypedPlutusCore
+import           Language.UntypedPlutusCore.DeBruijn
+
+import           "pure-zlib" Codec.Compression.Zlib  as PureZlib
+import           "zlib" Codec.Compression.Zlib       as Zlib
+import           Codec.Serialise                     (deserialise, serialise)
+import           Flat                                (DecodeException (..), flat, unflat)
+
+type Tm = Term Name DefaultUni ()
+
+data Codec a = Codec
+  { serialize   :: a -> BS.ByteString
+  , deserialize :: BS.ByteString -> a
+  }
+
+fromDecoded :: Show error => Either error a -> a
+fromDecoded (Left err) = error $ show err
+fromDecoded (Right  v) = v
+
+flatCodec :: Codec (Tm)
+flatCodec = Codec
+  { serialize   = force . flat
+  , deserialize = force . fromDecoded . unflat
+  }
+
+cborCodec :: Codec (Tm)
+cborCodec = Codec
+  { serialize   = force . LBS.toStrict . serialise
+  , deserialize = force . deserialise . LBS.fromStrict
+  }
+
+withZlib :: NFData a => Codec a -> Codec a
+withZlib codec = Codec
+  { serialize = force . LBS.toStrict . Zlib.compress . LBS.fromStrict . serialize codec
+  , deserialize = force . (deserialize codec) . LBS.toStrict . Zlib.decompress . LBS.fromStrict
+  }
+
+withPureZlib :: NFData a => Codec a -> Codec a
+withPureZlib codec = Codec
+  { serialize = serialize (withZlib codec)
+  , deserialize = force . (deserialize codec) . LBS.toStrict . fromDecoded . PureZlib.decompress . LBS.fromStrict
+  }
+
+codecs    :: [ (Text, Codec Tm) ]
+codecs    =
+  [ ("flat", Codec.flatCodec)
```

Comment:
Why are these all qualified?  They're all defined in this file and I don't think there are any clashes.  It's a bit confusing because there are things imported that begin with `Codec` as well.

Reply:
I actually have no good answer to that. I removed them.